client/blocks/edit-gravatar/index.jsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FilePicker from 'components/file-picker';
+import FormLabel from 'components/forms/form-label';
+import { getCurrentUser } from 'state/current-user/selectors';
+import {
+	isCurrentUserUploadingGravatar,
+} from 'state/current-user/gravatar-status/selectors';
+import Gravatar from 'components/gravatar';
+import { isOffline } from 'state/application/selectors';
+import { localize } from 'i18n-calypso';
+import Spinner from 'components/spinner';
+import { getToken as getOauthToken } from 'lib/oauth-token';
+import { uploadGravatar } from 'state/current-user/gravatar-status/actions';
+
+/**
+ * Module dependencies
+ */
+const debug = debugFactory( 'calypso:edit-gravatar' );
+
+export class EditGravatar extends Component {
+	constructor() {
+		super( ...arguments );
+		this.handleOnPick = this.handleOnPick.bind( this );
+	}
+
+	static propTypes = {
+		isOffline: PropTypes.bool,
+		isUploading: PropTypes.bool,
+		translate: PropTypes.func,
+		uploadGravatar: PropTypes.func,
+		user: PropTypes.object,
+	};
+
+	handleOnPick( files ) {
+		const { uploadGravatar: uploadGravatarAction, user } = this.props;
+		debug( 'you picked', JSON.stringify( files[ 0 ].name ) );
+
+		// check for bearerToken from desktop app
+		let bearerToken = getOauthToken();
+
+		if ( ! bearerToken ) {
+			bearerToken = localStorage.getItem( 'bearerToken' );
+		}
+
+		// send gravatar request
+		if ( bearerToken ) {
+			debug( 'Got the bearerToken, sending request' );
+			uploadGravatarAction( files[ 0 ], bearerToken, user.email );
+		} else {
+			debug( 'Oops - no bearer token.' );
+		}
+	}
+
+	render() {
+		const { isOffline: userIsOffline,
+			translate, user, isUploading } = this.props;
+		return (
+			<div>
+				<FormLabel>
+					{ translate( 'Avatar', {
+						comment: 'A section heading on the profile page.'
+					}
+					) }
```

Comment:
nit-pick: What do you think about placing the braces & parens on one line like so: `} ) }`?
I find that a tiny bit more readable since it better matches the opening braces & parens